Source/SwiftLintFramework/Rules/ColonRule.swift

```diff
@@ -41,19 +49,112 @@ public struct ColonRule: Rule {
             "func abc(def :Void) {}\n",
             "func abc(def : Void) {}\n",
             "func abc(def: Void, ghi :Void) {}\n"
+        ],
+        corrections: [
+            "let abc:Void\n": "let abc: Void\n",
+            "let abc:  Void\n": "let abc: Void\n",
+            "let abc :Void\n": "let abc: Void\n",
+            "let abc : Void\n": "let abc: Void\n",
+            "let abc : [Void: Void]\n": "let abc: [Void: Void]\n",
+            "let abc : (Void, String, Int)\n": "let abc: (Void, String, Int)\n",
+            "let abc : ([Void], String, Int)\n": "let abc: ([Void], String, Int)\n",
+            "let abc : [([Void], String, Int)]\n": "let abc: [([Void], String, Int)]\n",
+            "let abc:  (Void, String, Int)\n": "let abc: (Void, String, Int)\n",
+            "let abc:  ([Void], String, Int)\n": "let abc: ([Void], String, Int)\n",
+            "let abc:  [([Void], String, Int)]\n": "let abc: [([Void], String, Int)]\n",
+            "let abc :String=\"def\"\n": "let abc: String=\"def\"\n",
+            "let abc :Int=0\n": "let abc: Int=0\n",
+            "let abc :Int = 0\n": "let abc: Int = 0\n",
+            "let abc:Int=0\n": "let abc: Int=0\n",
+            "let abc:Int = 0\n": "let abc: Int = 0\n",
+            "let abc:Enum=Enum.Value\n": "let abc: Enum=Enum.Value\n",
+            "func abc(def:Void) {}\n": "func abc(def: Void) {}\n",
+            "func abc(def:  Void) {}\n": "func abc(def: Void) {}\n",
+            "func abc(def :Void) {}\n": "func abc(def: Void) {}\n",
+            "func abc(def : Void) {}\n": "func abc(def: Void) {}\n",
+            "func abc(def: Void, ghi :Void) {}\n": "func abc(def: Void, ghi: Void) {}\n"
         ]
     )
 
     public func validateFile(file: File) -> [StyleViolation] {
-        let pattern = "\\w+\\s+:\\s*\\S+|\\w+:(?:\\s{0}|\\s{2,})\\S+"
+        let pattern = (patterns() as NSArray).componentsJoinedByString("|")
```

Comment:
You could use `SequenceType`'s `joinWithSeparator` here to avoid the cast to `NSArray`.